libs/agno/agno/run/agent.py

```diff
@@ -489,6 +491,56 @@ def run_output_event_from_dict(data: dict) -> BaseRunOutputEvent:
     return cls.from_dict(data)  # type: ignore
 
 
+@dataclass
+class RunRequirement:
+    """Requirement to complete a paused run (used in HITL flows)"""
+
+    tool_execution: Optional[ToolExecution] = None
+    created_at: datetime = datetime.now(timezone.utc)
+
+    # User confirmation
+    needs_confirmation: bool = False
+    confirmation_note: Optional[str] = None
+    confirmation: Optional[bool] = None
+
+    # User input
+    needs_user_input: bool = False
+    user_input_schema: Optional[List[UserInputField]] = None
+    user_input: Optional[str] = None
+
+    # External execution
+    needs_external_execution: bool = False
+
+    def __init__(self, tool_execution: ToolExecution):
+        self.tool = tool_execution
+        self.needs_confirmation = tool_execution.requires_confirmation or False
+        self.needs_user_input = tool_execution.requires_user_input or False
+        self.user_input_schema = tool_execution.user_input_schema
+        self.needs_external_execution = tool_execution.external_execution_required or False
+
+    def answer(self, user_response: str):
```

Comment:
`give_input` rather?

Reply:
`requirement.answer(...)` looks sweet

Maybe `requirement.set_user_input(...)` is more clear?